### ISSUE TITLE
Persistent Unread Message indicator

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#2442]
### Problem

For some reason in the GIG group it's showing the `#general` chatroom as having an unread message for me:

![Screenshot 2024-11-30 at 5 17 38 PM](https://github.com/user-attachments/assets/65b70787-8c6a-4721-b4d0-f738199ce979) 

This is also causing a red bad to appear on the group icon in the tab menu (favicon) and in the app too:

![Screenshot 2024-11-30 at 5 23 01 PM](https://github.com/user-attachments/assets/f3fb6738-477d-422a-92bb-045c7ed09001)

It won't go away, even if I switch from `#general` to another chatroom and back and scroll all the way down.

EDIT: seeing this for DMs too on 1.2.0:

![Image](https://github.com/user-attachments/assets/3c841d8e-63eb-4cee-ba05-da58e760f07e)

### Solution

Find out why this is happening and fix.

**Comment by taoeffect** [comment]
FYI, the unread indicator went away after I posted a new message to the #general channel.

**Comment by corrideat** [comment]
This seems like a state inconsistency issue, where the unread message state in the browser is wrong. Sending a message triggered a KV reload. Likely restarting the service worker would have had the same effect (but not refreshing the page).

As for why the state was wrong to begin with, I'm not sure.

**Comment by dotmacro** [comment]
Possible steps to reproduce:
1. u1 sends message
2. u1 deletes message
[Edit to add: u2 now checks their unread messages]
3. u2 has persistent unread message indicator in dm/channel where u1 sent & deleted message
4. u2's unread message indicators disappear once u2 sends a message in the dm/channel with the unread message indicator

**Comment by SebinSong** [comment]
I was able to reproduce the issue from the steps described [above](https://github.com/okTurtles/group-income/issues/2442#issuecomment-4403460493) but _only when_ the message sent by `u1` was deleted while `u2` isn't logged in.

In case `u2` is logged in, clearing the deleted message from `u2`'s unread-message list is correctly handled in the `sideEffect` block of `gi.contracts/chatroom/deleteMessage` action:

https://github.com/okTurtles/group-income/blob/3d4d4bff273a7bc397d309b8d83621ca2e757bb1/frontend/model/contracts/chatroom.js#L524-L533

**Comment by taoeffect** [comment]
Awesome, that's progress I suppose! Being able to consistently reproduce the problem is one step from fixing it. 😄

**Comment by dotmacro** [comment]
> I was able to reproduce the issue from the steps described [above](https://github.com/okTurtles/group-income/issues/2442#issuecomment-4403460493) but _only when_ the message sent by `u1` was deleted while `u2` isn't logged in.

To clarify, do you mean that `u2` must actually log out (by clicking the `Logout` button on their settings page), or only that their app isn't actively open at that moment (e.g. device is asleep/locked)?

**Comment by SebinSong** [comment]
Yes, I had to log out by clicking on the log-out button to reproduce the issue.

**Comment by dotmacro** [comment]
I can reproduce without clicking the `Logout` button, with the app only being inactive (device asleep/locked). I'll re-test again later today to confirm. 
EDIT: confirmed!
